dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -550,15 +923,119 @@ public void test_MultilangContent_IndexingAllLanguages()
+
+            final Host site = new SiteDataGen().nextPersisted();
+            Language defaultLang = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language secondLang = new LanguageDataGen().nextPersisted();
+            folder = new FolderDataGen().site(site).nextPersisted();
+
+            Contentlet contentDefaultLang = createAndPublishEmployeeContent(site, defaultLang, "catherine");
+
+            // create container, template and htmlpage to reference previously created content
+
+            HTMLPageAsset page = createHtmlPageAsset(defaultLang, contentDefaultLang);
+
+            final String html = APILocator.getHTMLPageAssetAPI().getHTML(page.getURI(), site, true,
+                    contentDefaultLang.getIdentifier(), APILocator.systemUser(),
+                    contentDefaultLang.getLanguageId(), USER_AGENT_DOTCMS_SITESEARCH);
+
+            Assert.assertFalse(html.isEmpty());
+            Assert.assertTrue("Page should contain content's body", html.contains("catherine"));
+
+            final List<String> indicesBeforeTest = siteSearchAPI.listIndices();
+            for (final String index : indicesBeforeTest) {
+                esIndexAPI.delete(index);
+            }
+            final String jobId = UUIDUtil.uuid();
+            final String alias = "any-alias-" + System.currentTimeMillis();
+            final JobDataMap jobDataMap = new JobDataMap();
+            jobDataMap.put(SiteSearchJobImpl.RUN_NOW, Boolean.TRUE.toString());
+            jobDataMap.put(SiteSearchJobImpl.INCREMENTAL, Boolean.FALSE.toString());
+            jobDataMap.put(SiteSearchJobImpl.INDEX_ALIAS, alias);
+            jobDataMap.put(SiteSearchJobImpl.JOB_ID, jobId);
+            jobDataMap.put(SiteSearchJobImpl.QUARTZ_JOB_NAME,
+                    SiteSearchJobImpl.RUNNING_ONCE_JOB_NAME);
+            jobDataMap.put(SiteSearchJobImpl.INCLUDE_EXCLUDE, "all");
+            jobDataMap
+                    .put(SiteSearchJobImpl.LANG_TO_INDEX, new String[]{
+                            Long.toString(secondLang.getId())});
+            jobDataMap.put(SiteSearchJobImpl.INDEX_HOST, site.getIdentifier());
+
+            final JobDetail jobDetail = Mockito.mock(JobDetail.class);
+            Mockito.when(jobDetail.getJobDataMap()).thenReturn(jobDataMap);
+            final JobExecutionContext context = Mockito.mock(JobExecutionContext.class);
+            Mockito.when(context.getJobDetail()).thenReturn(jobDetail);
+            Mockito.when(context.getFireTime()).thenReturn(new Date());
+            final SiteSearchJobImpl impl = new SiteSearchJobImpl();
+            impl.run(context);
+
+            final List<String> indicesAfterTest = siteSearchAPI.listIndices();
+            Assert.assertFalse(indicesAfterTest.isEmpty());
+            final String newIndexName = indicesAfterTest.get(0);
+
+            SiteSearchResults searchResults = siteSearchAPI.search(newIndexName, "catherine", 0, 10);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'searchResults' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)